testing-common/src/test/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollectorTest.java

```diff
@@ -0,0 +1,408 @@
+    assertThat(references).containsKey("muzzle.TestClasses$MethodBodyAdvice$B");
+  }
+
+  @Test
+  public void shouldCreateReferencesForHelperClasses() {
+    ReferenceCollector collector = new ReferenceCollector(s -> false);
+    collector.collectReferencesFromAdvice(HelperAdvice.class.getName());
+    Map<String, ClassRef> references = collector.getReferences();
+
+    assertThat(references)
+        .containsOnlyKeys(
+            TestHelperClasses.Helper.class.getName(),
+            TestHelperClasses.HelperSuperClass.class.getName(),
+            TestHelperClasses.HelperInterface.class.getName());
+
+    ClassRef helperSuperClass = references.get(TestHelperClasses.HelperSuperClass.class.getName());
+    assertThat(helperSuperClass.getFlags()).contains(ManifestationFlag.ABSTRACT);
+    assertHelperSuperClassMethod(helperSuperClass, true);
+    assertMethod(
+        helperSuperClass,
+        "finalMethod",
+        "()Ljava/lang/String;",
+        VisibilityFlag.PUBLIC,
+        OwnershipFlag.NON_STATIC,
+        ManifestationFlag.FINAL);
+
+    ClassRef helperInterface = references.get(TestHelperClasses.HelperInterface.class.getName());
+    assertThat(helperInterface.getFlags()).contains(ManifestationFlag.ABSTRACT);
+    assertHelperInterfaceMethod(helperInterface, true);
+
+    ClassRef helperClass = references.get(TestHelperClasses.Helper.class.getName());
+    assertThat(helperClass.getFlags()).contains(ManifestationFlag.NON_FINAL);
+    assertHelperSuperClassMethod(helperClass, false);
+    assertHelperInterfaceMethod(helperClass, false);
+  }
+
+  @Test
+  public void shouldCollectFieldDeclarationReferences() {
+    ReferenceCollector collector =
+        new ReferenceCollector(s -> s.equals(DeclaredFieldTestClass.Helper.class.getName()));
+    collector.collectReferencesFromAdvice(DeclaredFieldTestClass.Advice.class.getName());
+    collector.prune();
+    Map<String, ClassRef> references = collector.getReferences();
+
+    ClassRef helperClass = references.get(DeclaredFieldTestClass.Helper.class.getName());
+    FieldRef superField = findField(helperClass, "superField");
+    assertThat(superField).isNotNull();
+    assertThat(superField.isDeclared()).isFalse();
+
+    FieldRef field = findField(helperClass, "helperField");
+    assertThat(field).isNotNull();
+    assertThat(field.isDeclared()).isTrue();
+
+    ClassRef libraryBaseClass =
+        references.get(DeclaredFieldTestClass.LibraryBaseClass.class.getName());
+    assertThat(libraryBaseClass.getFields()).isEmpty();
+  }
+
+  @Test
+  public void shouldFindAllHelperClasses() {
+    ReferenceCollector collector = new ReferenceCollector(s -> false);
+    collector.collectReferencesFromAdvice(HelperAdvice.class.getName());
+    collector.prune();
+    List<String> helperClasses = collector.getSortedHelperClasses();
+
+    assertThat(helperClasses)
+        .containsSubsequence(
+            Arrays.asList(
```

Comment:
Think you can use varargs instead of asList

https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/ObjectEnumerableAssert.html#containsSubsequence(ELEMENT...)